src/test/reporters/json.ts

```diff
@@ -216,13 +224,24 @@ class JSONReporter implements Reporter {
       annotations: test.annotations,
       expectedStatus: test.expectedStatus,
       projectName: test.titlePath()[1],
-      results: test.results.map(r => this._serializeTestResult(r)),
+      results: test.results.map(r => this._serializeTestResult(r, test.location.file)),
       status: test.outcome(),
     };
   }
 
-  private _serializeTestResult(result: TestResult): JSONReportTestResult {
+  private _serializeTestResult(result: TestResult, file: string): JSONReportTestResult {
     const steps = result.steps.filter(s => s.category === 'test.step');
+    let position: { column?: number; line?: number } = {};
+    if (file && result.error?.stack !== undefined) {
```

Comment:
I think we should not duplicate this logic, but rather put it into `positionInFile` and rename it to `prepareErrorStack(stack: string, file?: string): { stackLines: string[], position: ...}`